satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,503 @@
+    lat/lon as NaN as well.  The user can use the original lat/lon for those
+    pixels or use the higher level :class:`ParallaxCorrection` class.
+
+    This function assumes a spherical Earth.
+
+    Args:
+        sat_lon (number): Satellite longitude in geodetic coordinates [°]
+        sat_lat (number): Satellite latitude in geodetic coordinates [°]
+        sat_alt (number): Satellite altitude above the Earth surface [m]
+        lon (array or number): Longitudes of pixel or pixels to be corrected,
+            in geodetic coordinates [°]
+        lat (array or number): Latitudes of pixel/pixels to be corrected, in
+            geodetic coordinates [°]
+        height (array or number): Heights of pixels on which the correction
+            will be based.  Typically this is the cloud top height. [m]
+
+    Returns:
+        tuple[float, float]: New geolocation
+            New geolocation ``(lon, lat)`` for the longitude and
+            latitude that were to be corrected, in geodetic coordinates. [°]
+    """
+    # Be careful with units here.  Heights may be either in m or km, and may
+    # refer to either the Earth's surface on the Earth's centre.  Cloud top
+    # height is usually reported in metres above the Earth's surface, rarely in
+    # km.  Satellite altitude may be reported in either m or km, but orbital
+    # parameters may be in relation the the Earths centre.  The Earth radius
+    # from pyresample is reported in km.
+
+    elevation = _get_satellite_elevation(sat_lon, sat_lat, sat_alt, lon, lat)
+    parallax_distance = _calculate_parallax_distance(height, elevation)
+    shifted_xyz = _get_parallax_shift_xyz(
+            sat_lon, sat_lat, sat_alt, lon, lat, parallax_distance)
+
+    return xyz2lonlat(
+        shifted_xyz[..., 0], shifted_xyz[..., 1], shifted_xyz[..., 2])
+
+
+def _get_parallax_shift_xyz(sat_lon, sat_lat, sat_alt, lon, lat, parallax_distance):
+    """Calculate the parallax shift in cartesian coordinates.
+
+    From satellite position and cloud position, get the parallax shift in
+    cartesian coordinates.
+    """
+    sat_xyz = np.hstack(lonlat2xyz(sat_lon, sat_lat)) * sat_alt
+    cth_xyz = np.stack(lonlat2xyz(lon, lat), axis=-1) * EARTH_RADIUS*1e3  # km → m
+    delta_xyz = cth_xyz - sat_xyz
+    sat_distance = np.sqrt((delta_xyz*delta_xyz).sum(axis=-1))
+    dist_shape = delta_xyz.shape[:-1] + (1,)  # force correct array broadcasting
+    return cth_xyz - delta_xyz*(parallax_distance/sat_distance).reshape(dist_shape)
+
+
+def _get_satellite_elevation(sat_lon, sat_lat, sat_alt, lon, lat):
+    """Get satellite elevation.
+
+    Get the satellite elevation from satellite lon/lat/alt for positions
+    lon/lat.
+    """
+    placeholder_date = datetime.datetime(2000, 1, 1)  # no impact on get_observer_look?
+    (_, elevation) = get_observer_look(
+            sat_lon, sat_lat, sat_alt/1e3,  # m → km (wanted by get_observer_look)
+            placeholder_date, lon, lat, 0)
```

Comment:
Can this be replaced to make use of existing functions? Like this one https://github.com/pytroll/satpy/blob/4971b7cc49ecee9bfeeb1894ef38a3e3b02cb6f7/satpy/modifiers/angles.py#L416-L425